syntax = "proto3";
package raft;

service FollowerInteraction
{
	rpc AppendEntries(Append_info) returns(returnAppend){}//写在follower里，由leader调用
	rpc VoteRequest(VoteInfo) returns(returnVote){}//写在follower里，由candidate调用
}

message Append_info
{
	int64 term = 1;
	int32 leaderID = 2;
	int64 prevLogIndex = 3;
	int64 prevLogTerm = 4;
	string entry = 5;
	int64 leaderCommit = 6;
}

message returnAppend
{
	int64 term = 1;
	bool success = 2;
}

message VoteInfo
{
	int64 term = 1;
	int32 candidateID = 2;
	int64 lastLogIndex = 3;
	int64 lastLogTerm = 4;
}

message returnVote
{
	int64 term = 1;
	bool voteGranted = 2;
}
